docs/src/whats-available/crate-stores.md

```diff
@@ -4,12 +4,43 @@ Available crate stores
 'disk': Local on-disk store
 ---------------------------
 
-This store implements simple local storage of crates as files in a given directory.
+This strategy implements simple local storage of crates as files in a given directory.
 
-Here is an example of a configuration using this storage:
+Here is an example of configuration to use this storage strategy:
 
 ```toml
 [storage]
 type = "disk"           # required.
 path = "crate-storage"  # required: path of the directory in which to store the crates.
 ```
+
+'s3': AWS S3 object storage
+---------------------------
+
+This strategy stores crate archives and READMEs as objects within an AWS S3 bucket.
+
+Here is an example of configuration to use this storage strategy:
+
+```toml
+[storage]
+type = "s3"                     # required.
+region = ["eu-west-1"]          # required: name of the operating region of the S3 bucket.
+bucket = "eu-polomack-crates"   # required: name of the S3 bucket to use.
+key_prefix = "crates"           # optional: arbitrary prefix to apply on the objects' keys
+                                #           allowing to place them in subdirectories.
+```
+
+You can specify a custom S3 endpoint, instead of the official S3 ones, using the `region` key, like this:
+
+```toml
+region = ["custom", "https://my.custom.s3.endpoint/"]
+```
+
+### S3 Authentication
+
+In order to authenticate the registry to S3, you can either:
+
+- define both `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables when running the registry.
+- have credentials stored in `~/.aws/config` (called the 'profile') and giving the registry read permission to it.
```

Comment:
The [rusoto docs](https://docs.rs/rusoto_credential/0.44.0/rusoto_credential/struct.ChainProvider.html) for their default provider (which is what we use) list four options, checked in this order:

> 1. Environment variables: AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
> 2. credential_process command in the AWS config file, usually located at ~/.aws/config.
> 3. AWS credentials file. Usually located at ~/.aws/credentials.
> 4. IAM instance profile. Will only work if running on an EC2 instance with an instance profile/role.

I _think_ 2 and 4 are relatively rare compared to 1 and 3 (at least, I've personally used 1 and 3 but never 2 or 4). I'm not sure how much we want to duplicate their docs though. Maybe include a link to theirs? If we keep this more or less as-is, I think we should change to "have credentials stored in `~/.aws/credentials`".

Reply:
I had included the two options that I used the most as well (1 and 3), but I think it should be ok to mention all 4 options in here, they're quite short to mention in full.  
I also included Rusoto's documentation link for more details if required.